qcfractal/interface/collections/collection.py

```diff
@@ -132,36 +147,13 @@ def from_json(cls, data, client=None):
             raise KeyError("Attempted to create Collection from JSON with class {}, but found collection type of {}.".
                            format(class_name, data["collection"]))
 
-        # Ensure all required fields are found.
-        req_fields = cls.__base_fields | getattr(cls, "_" + cls.__name__ + "__required_fields")
-        if len(req_fields - data.keys()):
-            missing = req_fields - data.keys()
+        # Attempt to build class, ensure all fields to reconstruct from JSON are present
+        req_fields = cls.DataModel.schema()['properties'].keys()
+        missing = req_fields - data.keys()
+        if len(missing):
```

Comment:
Can we make these required and it is autovalidated via PyDantic?

Reply:
> Can we make these required 

Many of the keys have default values, which is something we want in places, having a default automatically makes it `optional` from a PyDantic standpoint. However, this is a special case where we want data *fully defined* when being fed back in from the server, even if that data was default value when originally created. This preserves check traps the case of rebuilding the object from existing data where some fields were omitted.

We could remove this check and allow missing fields to have their defaults filled in. Mandatory fields will still be checked by PyDantic

>  autovalidated via PyDantic?

Types and mandatory fields will still be checked, but this is a different type of check.

Reply:
Good point, can you slightly enhance the comment line slightly to describe why we are approaching it this way. Not a big issue.

Reply:
Will do!